### ISSUE TITLE
I can never stitch anything. Just the same error

### ISSUE THREAD

**Issue by tcdoe** [fiji/Stitching#63]
Hi,
I've tried many variations and followed all instructions. However all I ever get is this error (below at end).
My images are taken in very standard way just moving X (column) from left to right, and then Y (next row).
Filenames are typical:   testneur3d10x_L01_R01_C01.jpg,   testneur3d10x_L01_R01_C02.jpg,  etc. 

I have tried the instructed file name notations variations:
testarr_L01_R{yy}_C{xx}.jpg
testarr_L01_R{jj}_C{ii}.jpg
testarr_L{zz}_R{yy}_C{xx}.jpg

my filenames are as specifically all like this:  "testarr_L01_R01_C01.jpg"
where L is the z level, R is the row (y move) and C is the column (x move).

I have tried using {ii}, and {jj} etc;  but that does not work either. Throws basically similar error.
I have the initial index set to 1. 

Any help is much appreciated.

**Cannot find file:**
error:
Loading (0, 0): C:\AbemScan_v08\bin\x64\Debug\scan3D10xneuron02\testneur3d10x_L01_R{yy}_C{xx}.jpg ... 
ERROR: Cannot find file: 'C:\AbemScan_v08\bin\x64\Debug\scan3D10xneuron02\testneur3d10x_L01_R{yy}_C{xx}.jpg' - abort stitching.
ERROR: Error during tile discovery, or invalid grid type. Aborting.

**Comment by StephanPreibisch** [comment]
Hi @tcdoe, can you please post screenshots of the GUI that you use? It could be that you are maybe selecting the wrong options ...

**Comment by tcdoe** [comment]
shure,
here's one where I just use the xx and yy
https://i.imgur.com/8xejydZ.png
and then one with zz, yy, xx. 
https://i.imgur.com/lRyXkoi.png
funny thing is that i've done this before and had it working no prob, that was a couple years ago.

**Comment by tcdoe** [comment]
of course, the files are there. I tried putting them into a generic C:\test folder and same error.

**Comment by StephanPreibisch** [comment]
Hi @tcdoe, you select the wrong mode (Grid: row-by-row: Right & Down). You need to select "Positions from file" instead, you can this described in detail in the documentation (https://imagej.net/plugins/image-stitching#grid-collection-stitching). Then {xx} and {yy} will work.

Best,
Stephan

**Comment by tcdoe** [comment]
Hi, Thanks very much for reply! but that doesn't seem to work quite as you describe.

If I use Positions from File, or Filename defined Position, and put in:
{xx} and {yy}, the Filename defined position works for 2D, e.g. only for {xx} and {yy}, but **not for {zz}.**

I get the same file not found error. 
I will continue to try different variations of the TileConfiguration.txt file

Do I really have to hand-make a custom TileConfiguration.txt file to get this to work in 3D?
It seems extremely tedious, and nothing in the documentation indicates that. But if I have to, no problem it will at least be good to know that that is the only way.

I get this working good for 2 dimensions, but dim=3 hasn't worked so far.

# Define the number of dimensions we are working on
dim = 2

# Define the image coordinates
testneur3d10x_L01_R01_C01.jpg; ; (0.0, 0.0)
testneur3d10x_L01_R01_C02.jpg; ; (2406.0, 0.0)
testneur3d10x_L01_R01_C03.jpg; ; (4812.0, 0.0)
testneur3d10x_L01_R02_C01.jpg; ; (0.0, 2400.0)
testneur3d10x_L01_R02_C02.jpg; ; (2406.0, 2400.0)
testneur3d10x_L01_R02_C03.jpg; ; (4812.0, 2400.0)
testneur3d10x_L01_R03_C01.jpg; ; (0.0, 4800.0)
testneur3d10x_L01_R03_C02.jpg; ; (2406.0, 4800.0)
testneur3d10x_L01_R03_C03.jpg; ; (4812.0, 4800.0)

**Comment by StephanPreibisch** [comment]
Hi, for dim=3 you usually use 3D images (e.g. a 3D TIFF) and not slices ... maybe that is the problem?

**Comment by tcdoe** [comment]
Hi Thanks, I just have separate images in tiles; each image is a jpg. 
So you can think of it as what I thought it would be a simple extension of 2D image "tiles", eg: testscan_Z{zz}_Y{yy}_X{xx}.jpg
But I think now I understand that I would have to build the Z levels into such as a 3D TIFF, and then 'stich' those together.

Again, thanks very much for your help. I guess I'll try the above, and see what I can get.
I've made a 3D scanning microscope so now I'm trying to generate simple volumetric images from XYZ scanned tiles.
Best!

**Comment by StephanPreibisch** [comment]
Hi, as a first step I'd resave them to a 3D Tiff using Fiji and try if it actually works ... that is straight forward. Extending Stitching is always possible, but somebody needs to do it :)

**Comment by tcdoe** [comment]
Thanks will do. Fortunately I got the 3D Tiff to generate, still working on the rest. Boosting my memory to 128G hope that's enough :))
